isic_archive/models/image.py

```diff
@@ -29,16 +28,26 @@ def initialize(self):
 
         self.prefixSearchFields = ['lowerName', 'name']
 
+    @staticmethod
+    def generateIsicID():
+        taken = True
+        id_: int = 0
+
+        while taken:
+            id_ = secrets.randbelow(10_000_000)
+            taken = bool(Image().findOne({'name': f'ISIC_{id_}'}).count())
+
+        return f'ISIC_{id_}'
```

Comment:
* Need to string-format the new value to ensure leading zeros.

* `findOne` returns a `dict` or `None`, so you need to use `find` to work with `.count()`. In general, `.find().count()` is preferred to `.findOne()`, since the actual document is not returned.

```suggestion
			isicId = f'ISIC_{id_:07d}'
            taken = Image().find({'name': isicId}).count() > 0

        return isicId
```